runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -317,7 +317,10 @@ private void hideWarnedStyles(boolean enabled)
 			}
 
 			// Magic staves defensive casting mode
-			if (equippedWeaponType == WeaponType.TYPE_18)
+			// Note: Staff of  the Dead is weapon type 21, rather than the expected type 18
+			if (equippedWeaponType == WeaponType.TYPE_18
+				|| equippedWeaponType == WeaponType.TYPE_21
+				|| !enabled)
```

Comment:
Why was this !enabled line added?

Reply:
It's part of the second commit, which is to fix the issue of the defensive casting style being permanently hidden if you have a non-staff equipped and disable the plugin or disable style hiding.